feat: add selectOnFocus property to toggle selecting tabs on focus when arrow keys are pressed

StephenRiosDev opened the pull request:
feat: add selectOnFocus property to toggle selecting tabs on focus when arrow keys are pressed

# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Enhance the AuroTabgroup component with a new selectOnFocus option for keyboard navigation and programmatic selection capabilities while cleaning up internal event handlers and updating documentation

New Features:
- Add selectOnFocus boolean property to AuroTabgroup to optionally select tabs on focus during arrow-key navigation
- Introduce selectTabByIndex method to programmatically select a tab by its index
- Enable selecting an AuroTab with Enter or Space keys via added keydown listener

Enhancements:
- Rename internal slider styling handler to handleTabSelected and update its event bindings
- Update API documentation to include selectOnFocus property and new methods

StephenRiosDev committed:
feat: add selectOnFocus property to toggle selecting tabs on focus when arrow keys are pressed

StephenRiosDev commented:
@sourcery-ai review

StephenRiosDev commented on src/auro-tabgroup.js:
```diff
@@ -261,7 +271,7 @@ export class AuroTabgroup extends LitElement {
   addEventListeners() {
-    this.addEventListener("tab-selected", this.setSliderStyles);
+    this.addEventListener("tab-selected", this.handleTabSelected);
     this.addEventListener("keydown", this.onKeyDown);
```
Enter and space keydown is bound in the AuroTab component. Confirmed pressing enter or space successfully selects tab. Was there another issue here that I'm missing?

sun-mota commented on src/auro-tabgroup.js:
```diff
@@ -50,6 +50,15 @@ export class AuroTabgroup extends LitElement {
         type: Number,
```
and also docs!

StephenRiosDev replied:
We need to audit tabs for better docs and unit testing. It's part of tech debt. After discussing with Doug we decided it's ok if we pass on this for now in order to support releasing this by EOD today,

sun-mota commented:
changing value of seleconfocus does not work correctly. Did I do something wrong here? 
I didnt use Tab but only left/right in this vdieo.

https://github.com/user-attachments/assets/9d353a09-02de-457a-8dc1-779a4d86c247

StephenRiosDev commented:
> changing value of seleconfocus does not work correctly. Did I do something wrong here? I didnt use Tab but only left/right in this vdieo.
> 
>  Screen.Recording.2025-08-20.at.12.40.21.PM.mov

The `selectOnFocus` property/attribute is boolean only, it doesn't take a string true/false. This requires a custom converter to support in Lit. This means anytime the attribute exists it's true regardless of the value assigned. 

Case 1: Host element has the `selectOnFocus` attribute
Result: Tabs are selected automatically when hitting in the arrows

Case 2: Host element does not have `selectOnFocus` attribute
Result: Tab focus changes but are not selected until the user hits enter or space

sun-mota commented:
oh... actually, it works as expected. it was just the styling was confusing to distinguish focus vs select. lol

> > changing value of seleconfocus does not work correctly. Did I do something wrong here? I didnt use Tab but only left/right in this vdieo.
> > Screen.Recording.2025-08-20.at.12.40.21.PM.mov
> 
> The `selectOnFocus` property/attribute is boolean only, it doesn't take a string true/false. This requires a custom converter to support in Lit. This means anytime the attribute exists it's true regardless of the value assigned.
> 
> Case 1: Host element has the `selectOnFocus` attribute Result: Tabs are selected automatically when hitting in the arrows
> 
> Case 2: Host element does not have `selectOnFocus` attribute Result: Tab focus changes but are not selected until the user hits enter or space

StephenRiosDev committed:
fix: better focus handling for keyboard controls #75

- Focusing a tab now changes it's tabindex so we maintain the user's focus during tab operations
- Tabgroup now handles setting the focused state of all tabs when a tab is focused
- Added event for tab-blurred
